### ISSUE TITLE
Re

### ISSUE THREAD

**Issue by duansanshi** [FMLYD/psw-i#1]
all_mask=np.isnan(X)
val=X
train=mcar(X,p=0.2)
mask = np.isnan(train) 
val_mask=np.isnan(train) ^(np.isnan(X_all)) 
Is it the case that val_mask contains all_mask, and if so, would this lead to a data leakage problem when using the error on val_mask as the stopping criterion? I hope the author can provide a response, thank you very much!

**Comment by duansanshi** [comment]
Executing each shell script still does not yield the results described in the paper.

**Comment by FMLYD** [comment]
Sorry for my late reply.

I recently discovered that the issue with the traffic.csv file might have occurred when I compressed the entire folder and uploaded it to GitHub. I’m not exactly sure how this problem happened, especially since this repository was uploaded about four months ago. As a result, the traffic.csv file now only contains around 200 lines, which is different from the original version. You can download the correct version of the dataset from the iTransformer repository.

Regarding the code, I am confident that it does not introduce data leakage. Specifically, the line:
val_mask = np.isnan(train) ^ (np.isnan(X_all))

ensures that val_mask retains only the values that are True in train but False in X_all. If a value is True (i.e., missing) in both train and X_all, it is set to False in val_mask. This mechanism effectively prevents data leakage because val_mask does not include missing values present in both the original dataset (X_all) and the training set (train). Instead, it only captures newly introduced missing values.

Please note that the current code in this repository was temporarily uploaded during the rebuttal period. The final version of the code is still being cleaned up and will be released along with the experimental logs submitted before the paper’s final submission.

**Comment by FMLYD** [comment]
Thank you very much for your interest in our work. If you have any further questions, please don’t hesitate to reach out to me.

**Comment by duansanshi** [comment]
Thank you for your kind answering, but I still believe there is a data leakage issue in the current implementation. Here's the detailed analysis:

X is obtained by randomly masking the original data X_all, with **all_mask** identifying the missing values in X.

Then, train is created by further adding 20% missing values to X, with **mask** identifying these missing parts.

The current **mask includes all missing values from all_mask**, meaning where all_mask is 1, mask is also 1.

Subsequently, the created val_mask also includes all_mask (except for the parts missing in X_all), which leads to data leakage.

To prevent data leakage, val_mask should be calculated as np.isnan(train) ^ np.isnan(X) instead of using X_all.

Is my analysis correct? Please point out if there are any mistakes.

**Comment by FMLYD** [comment]
Thank you for your reminder — you’re absolutely right, and I truly appreciate you catching this mistake. During the rebuttal period, I mistakenly uploaded an early experimental folder that was used for overfitting tests. Thank you once again for pointing this out. We’ll improve our file organization in the future to avoid such oversights.
